src/products/components/ProductListDatagrid/ProductListDatagrid.tsx

```diff
@@ -0,0 +1,255 @@
+
+      if (canBeSorted(columnName, !!selectedChannelId)) {
+        onSort(columnName, columnId);
+      }
+    },
+    [columns, onSort, selectedChannelId],
+  );
+
+  const handleRowClick = useCallback(
+    ([_, row]: Item) => {
+      const rowData = products[row];
+      onRowClick(rowData.id);
+    },
+    [onRowClick, products],
+  );
+
+  const handleGetColumnTooltipContent = useCallback(
+    (colIndex: number): string => {
+      const { columnName } = getColumnMetadata(columns[colIndex].id);
+      // Sortable column
+      if (canBeSorted(columnName, !!selectedChannelId)) {
+        return "";
+      }
+      // No sortable column
+      if (!Object.keys(ProductListUrlSortField).includes(columnName)) {
+        return intl.formatMessage(commonTooltipMessages.noSortable);
+      }
+
+      // Sortableb but required selected channl
+      return intl.formatMessage(commonTooltipMessages.noFilterSelected, {
+        filterName: filterDependency.label,
+      });
+    },
+    [columns, filterDependency.label, intl, selectedChannelId],
+  );
+
+  const onColumnsChange = useCallback(
+    (columns: ProductListColumns[]) => onUpdateListSettings("columns", columns),
+    [onUpdateListSettings],
+  );
+
+  return (
+    <DatagridChangeStateContext.Provider value={datagrid}>
+      <Datagrid
+        readonly
+        rowMarkers="none"
+        columnSelect="single"
+        freezeColumns={2}
+        verticalBorder={col => (col > 1 ? true : false)}
+        getColumnTooltipContent={handleGetColumnTooltipContent}
+        availableColumns={columns}
+        onHeaderClicked={handleHeaderClicked}
+        emptyText={intl.formatMessage(messages.emptyText)}
+        getCellContent={getCellContent}
+        getCellError={() => false}
+        menuItems={() => []}
+        rows={getProductRowsLength(disabled, products)}
+        selectionActions={(indexes, { removeRows }) => (
+          <Button variant="tertiary" onClick={() => removeRows(indexes)}>
+            <FormattedMessage {...buttonMessages.delete} />
+          </Button>
+        )}
+        title=""
+        fullScreenTitle={intl.formatMessage(messages.products)}
+        onChange={onChange}
+        onRowClick={handleRowClick}
+        renderColumnPicker={defaultProps => (
+          <ColumnPicker
+            {...defaultProps}
+            {...columnPickerColumns}
+            hasMore={hasMore}
+            loading={loading}
+            onFetchMore={onFetchMore}
+            query={columnQuery}
+            onQueryChange={onColumnQueryChange}
+            onSave={onColumnsChange}
+          />
+        )}
+      />
+
+      <div className={classes.paginationContainer}>
+        <TablePaginationWithContext
+          component="div"
+          colSpan={(products?.length === 0 ? 1 : 2) + settings.columns.length}
+          settings={settings}
+          disabled={disabled}
+          onUpdateListSettings={onUpdateListSettings}
+        />
+      </div>
+      {isDirty && (
```

Comment:
**question**: will it currently be true?

Reply:
No, I've prepared first editable version. I left this for future when we will be working on edit mode.